test/e2e/e2e_test.go

```diff
@@ -0,0 +1,204 @@
+	return wait.PollImmediateWithContext(ctx, 5*time.Second, 120*time.Second, func(ctx context.Context) (bool, error) {
+		promRes := schema.GroupVersionResource{Group: "monitoring.coreos.com", Version: "v1", Resource: "prometheuses"}
+		prom, err := client.Resource(promRes).Namespace(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		var condReconciled string
+		var condAvailable string
+		if conditions, found, err := unstructured.NestedSlice(prom.Object, "status", "conditions"); err != nil {
+			return false, err
+		} else if found {
+			for _, condition := range conditions {
+				c := condition.(map[string]interface{})
+				if c["type"] == "Reconciled" {
+					condReconciled = c["status"].(string)
+				} else if c["type"] == "Available" {
+					condAvailable = c["status"].(string)
+				}
+			}
+		}
+
+		if condReconciled != "True" {
+			t.Logf("Prometheus instance '%s': Waiting for reconciliation...", name)
+			return false, nil
+		}
+
+		specReplicas, found, err := unstructured.NestedInt64(prom.Object, "spec", "replicas")
+		if !found {
+			return false, errors.New("no .spec.replicas")
+		} else if err != nil {
+			return false, err
+		}
+
+		availableReplicas, found, err := unstructured.NestedInt64(prom.Object, "status", "availableReplicas")
+		if !found {
+			return false, errors.New("no .status.availableReplicas")
+		} else if err != nil {
+			return false, err
+		}
+
+		if specReplicas != availableReplicas {
+			t.Logf("Prometheus instance '%s': %v/%v pods are ready. Waiting for all pods to be ready...", name, availableReplicas, specReplicas)
+			return false, err
+		}
+
+		if condAvailable != "True" {
+			t.Logf("Prometheus instance '%s': Available = %v. Waiting for Available status...", name, condAvailable)
+			return false, nil
+		}
+
+		t.Logf("Prometheus instance '%s': Ready.", name)
+		return true, nil
+	})
+}
+
+func waitForDeploymentReady(ctx context.Context, t *testing.T, clientSet clientset.Interface, namespace string, name string) error {
+	return wait.PollImmediateWithContext(ctx, 5*time.Second, 30*time.Second, func(ctx context.Context) (bool, error) {
+		sts, err := clientSet.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if sts.Status.ReadyReplicas == *sts.Spec.Replicas {
+			t.Logf("Deployment %s: %v/%v pods are ready.", name, sts.Status.ReadyReplicas, *sts.Spec.Replicas)
+			return true, nil
+		}
+		t.Logf("Deployment %s: %v/%v pods are ready. Waiting for all pods to be ready...", name, sts.Status.ReadyReplicas, *sts.Spec.Replicas)
+		return false, nil
+	})
+}
+
+func TestNodeMetrics(t *testing.T) {
+	_, metricsClient := initializeClient(t)
+
+	ctx := context.Background()
+	var nodeMetrics *metricsv1beta1.NodeMetricsList
+	err := wait.PollImmediateWithContext(ctx, 5*time.Second, 120*time.Second, func(ctx context.Context) (bool, error) {
```

Comment:
The e2e cluster should be ready by that time so we should do more active polling with a lower timeout